src/MapperInterface.php

```diff
@@ -19,5 +19,5 @@ interface MapperInterface
      *
      * @return mixed The mapped value
      */
-    public function &map($value, array $context = []);
+    public function &map(mixed $value, array $context = []): mixed;
```

Comment:
isn't this dangerous?
since this interface is implemented in user land in generated mappers.
If hot reload is disabled couldn't some problem occur?

https://3v4l.org/JtHhF

Reply:
I always expect users to reload their transformers from one version to another that's why I think this shouldn't be an issue.

Reply:
Couldn't hot reload strategy be based on the library version?
I've never checked this part of code since I've based my code on the cache warmer

Reply:
Couldn't hot reload strategy be based on the library version?
I've never checked this part of code since I've based my code on the cache warmer

Reply:
Today the hot reload is based onto a hash of the transformer metadata. We totally could add the AutoMapper version into this to make it force reload if the version doesn't match :100: